dowhy/causal_refuters/dummy_outcome_refuter.py

```diff
@@ -121,10 +133,19 @@ def refute_estimate(self):
 
         simulation_results = []
         refute_list = []
-        no_estimator = self.check_for_estimator()
+        
+        # We use collections.OrderedDict to maintain the order in which the data is stored
+        causal_effect_map = OrderedDict()
 
+        # Check if we are using an estimators in the transformation list 
+        no_estimator = self.check_for_estimator()
```

Comment:
I can understand the source of confusion. It rises from the fact that we have made use of a variable called no estimator. Thus, if we find no estimators in the transformation list. The value should be True, otherwise, the value should be false.
```
for action,_ in self._transformation_list:
            if callable(action) or action in DummyOutcomeRefuter.SUPPORTED_ESTIMATORS:
                return False
            
        return True
```

Reply:
I see. I still think that using a different variable name is better. "estimator" is overloaded here. An alternative is "learnable_transformation" since that's what we really want to check. Another name for the binary flag is "needs_train_set", which also conveys the logic.

Reply:
And the True/False oddity is because the name of the function is check_for_estimator. If you want the behavior you mentioned, then the function should be called check_for_no_estimator or something like that. It is weird that a function called check_for_estimator returns False when there is an estimator available. 
Anyway I think the binary flag variable needs to be changed to simply "has_estimator" (and negated in an if condition later).  double negatives are never easy to understand.

Reply:
cool